mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -864,6 +864,11 @@ trait Params extends Identifiable with Serializable {
     extractParamMap(ParamMap.empty)
   }
 
+  /**
+   * Extracts the user-supplied params.
+   */
+  final def extractUserParamMap(): ParamMap = paramMap
```

Comment:
can't we just make `paramMap` `private[ml]`?

Reply:
In this way I think you can also avoid the MiMa failure...

Reply:
Looks like it still can't avoid the MiMa failure.